go-controller/pkg/util/multi_network.go

```diff
@@ -98,7 +98,7 @@ func (nInfo *DefaultNetInfo) CompareNetInfo(netBasicInfo BasicNetInfo) bool {
 
 // TopologyType returns the defaultNetConfInfo's topology type which is empty
 func (nInfo *DefaultNetInfo) TopologyType() string {
-	return ""
+	return types.Layer3Topology
```

Comment:
I did look at usage in the code. I was mainly looking at where using TopologyType would break primary network if the value wasnt empty. Most of the usage for TopologyType is guarded by IsSecondary. I think it is safe to make this change now. I can clean up your example if I push another change, or if you require it to merge.